### PR TITLE
Embedding Looker reports in web application

### PR DESCRIPTION
**Pull request by taher435**
Adding article about how we are using looker embeds internally for interactive reports without having to use or build a charting library.

### CODE REVIEW

**Comment by so0k** [comment]
wow, cool, we had about once a month blog posts for a while - then it dropped again - thanks for this @taher435 - reviewing now

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+tags: [elixir,looker,reporting]
+author: Taher Dhilawala
+---
+## Background
+
+At [honestbee](https://honestbee.com), our BI (Business Intelligence) team uses [Looker](https://looker.com/) for analyzing data and build a visual graphical interface on top of that.
+Looker has proved to be a great asset for us since it allows our BI team to be independent of Engineering team. It helps our business teams drive decisions based on the data reports generated by Looker
+
+In all our different verticals (groceries, food, goodship - our B2B parcel business) we work closely with merchant partners.
+Earlier this year, we started working on a portal that allows our merchant partners to see how is their business performing on honestbee.
```
Comment (so0k) [inline _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: ok to share this info?

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+To do this, we had to build certain reports (with graphs) to help them understand things like their total sales, their busiest time of the day, their top selling products etc.
+To build these reports, we had all the data but to make sense of that data we would have to build a (or customise existing) graph library. For example, [Google charts](https://developers.google.com/chart/) or [Fusion Charts](https://www.fusioncharts.com/).
+This would increase our overall development time, which means, more time to go-to-market and get feedback from merchants.
+
+## Looker Embed
+
+Luckily, Looker provides a way to embed the already generated reports (looks and dashboards) in your web application. You can read a bit more about it [here](https://discourse.looker.com/t/creating-a-proof-of-concept-embedded-dashboard-powered-by-looker/824).
+
+The only problem was, we already had all the users and their roles and permission data with us and we didn't want to duplicate that data by creating groups again in looker.
```
Comment (so0k) [inline _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: seems the problem is more that `looker embed` requires you to add users / partners permissions into Looker and then...

> we already had all the users ...

Comment (so0k) [reply _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: seems that is described in the last paragraph of the article you linked?

>Use of Looker's SSO embedding option can allow embedding of dashboards to a set of specified external users. Your application would manage the creation and permissioning users in a programatic and scalable way, removing the requirement of creating and maintaining these users in Looker.

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Our service module in our API is responsible for reading all the user attributes and necessary filters and generate an SSO URL which is then passed to the frontend application.
+The SSO URL is then embedded in an iframe. Looker reports are interactive, which means, users can enable or disable a section of the pie chart. They can also download the data as PDF
+
+You can find a sample Elixir code ([since we use Elixir with Phoenix](http://tech.honestbee.com/articles/elixir/2017-04/enums-in-elixir-ecto)) that generates the SSO URL [here](https://github.com/taher435/looker_embed_sso_examples/tree/master/elixir).
+There are examples for other languages as well.
+
+## Summary
+
+With this setup, it only takes us few hours to release a new report to our merchant partners. Once our BI team is ready with a dashboard in looker, they configure the report's embed URL in the admin UI and that report goes live.
+It also allowed us to quickly build this merchant portal without having to spend a lot of time and effort on charting library and build features like export to PDF.
```
Comment (so0k) [inline _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: > a lot of time and effort on charting library

you [accidentally a word](http://knowyourmeme.com/memes/i-accidentally)?

> a lot of time and effort implementing ... ?

**Comment by so0k** [comment]
it would be really nice to have an image to show some of it... can be blurred / anonymised?

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md (hunk 1 above)
Comment (so0k) [reply _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: > how is their business performing on honestbee 

maybe?

> how their business is performing on honestbee

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+author: Taher Dhilawala
+---
+## Background
+
+At [honestbee](https://honestbee.com), our BI (Business Intelligence) team uses [Looker](https://looker.com/) for analyzing data and build a visual graphical interface on top of that.
+Looker has proved to be a great asset for us since it allows our BI team to be independent of Engineering team. It helps our business teams drive decisions based on the data reports generated by Looker
+
+In all our different verticals (groceries, food, goodship - our B2B parcel business) we work closely with merchant partners.
+Earlier this year, we started working on a portal that allows our merchant partners to see how is their business performing on honestbee.
+To do this, we had to build certain reports (with graphs) to help them understand things like their total sales, their busiest time of the day, their top selling products etc.
```
Comment (so0k) [inline _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: > to help them understand things like ...

what about?

> to provide them insights into things such as ...

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Looker Embed
+
+Luckily, Looker provides a way to embed the already generated reports (looks and dashboards) in your web application. You can read a bit more about it [here](https://discourse.looker.com/t/creating-a-proof-of-concept-embedded-dashboard-powered-by-looker/824).
+
+The only problem was, we already had all the users and their roles and permission data with us and we didn't want to duplicate that data by creating groups again in looker.
+
+## User Attributes for access control
+
+Looker allows us to [create attributes in looker](https://docs.looker.com/admin-options/settings/user-attributes) that applies to all the users created.
```
Comment (so0k) [inline _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: > Looker allows us to create attributes in looker that applies to all the users created.

how about

> Looker allows us to [add attributes](https://docs.looker.com/admin-options/settings/user-attributes) which can be applied to users or groups

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md (hunk 1 above)
Comment (taher435) [reply _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: I guess so. UberEats have published about their restaurant merchant portal - https://techcrunch.com/2017/03/02/ubereats-debuts-restaurant-manager-to-give-food-businesses-actionable-data/

**Comment by taher435** [comment]
Thanks @so0k. You are a great reviewer 👏 . Address all of your comments.

File: _posts/2017-11-07-embedding-looker-reports-in-web-app.md (hunk 1 above)
Comment (taher435) [reply _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: May be @ceik can help here 😄

Comment (so0k) [reply _posts/2017-11-07-embedding-looker-reports-in-web-app.md]: whoops I published it

**Pull request merged**